lib/api/1.1/southbound/files.js

```diff
@@ -71,6 +71,59 @@ function filesRouterFactory (
     });
 
     /**
+     * @api {get} /api/1.1/files/:filename/latest GET /:filename
+     * @apiVersion 1.1.0
+     * @apiDescription get the most recent version of a file by filename
+     * @apiName files-get
+     * @apiGroup files
+     * @apiError NotFound The file with the <code>uuid</code> was not found.
+     * @apiErrorExample Error-Response:
+     *     HTTP/1.1 404 Not Found
+     *     {
+     *       "error": "File not found."
+     *     }
+     */
+
+    router.get('/files/:filename/latest', function(req, res) {
+        logger.debug("Received request for file by uuid", {
+            uuid: req.params.uuid
+        });
+
+        var query = req.params.filename;
+
+        return fileService.verify(query)
+        .then(function(metadata) {
+            return metadata[0].uuid;
```

Comment:
Change to metadata[metadata.length - 1] to return the latest file. Similar to https://github.com/RackHD/on-http/pull/222